cmake/KokkosConfigCommon.cmake.in

```diff
@@ -306,3 +306,49 @@ FUNCTION(kokkos_compilation)
         ENDFOREACH()
     ENDIF()
 ENDFUNCTION()
+
+# this macro enables the cmake language used by  Kokkos:
+#
+#
+MACRO(kokkos_enable_language)
+  enable_language(${Kokkos_COMPILE_LANGUAGE})
+ENDMACRO()
+
+# this function is provided to easily set which files use the same programming language as Kokkos:
+#
+#       SOURCE      --> specific source files
+#
+FUNCTION(kokkos_set_source_language LANG_SOURCE)
+
+      IF(NOT DEFINED Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
+        MESSAGE(FATAL_ERROR "Using language features requires Kokkos to be built with Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE=ON")
+      ENDIF()
+      
+      IF(EXISTS ${LANG_SOURCE})
```

Comment:
This doesn't wotrk since you often don't get a full path. I'm seeing in the example
```
CMake Warning at /app/kokkos/build/KokkosConfigCommon.cmake:331 (MESSAGE):
  KOKKOS: SOURCE `cmake_example.cpp` could not be found - Source language
  could not be set
Call Stack (most recent call first):
  CMakeLists.txt:15 (kokkos_set_source_language)
```
If I remove the check, everything is fine.

Reply:
```diff
root@waffle01:/app/kokkos# git diff
diff --git a/cmake/KokkosConfigCommon.cmake.in b/cmake/KokkosConfigCommon.cmake.in
index ea1cc04a7..bb6264e4e 100644
--- a/cmake/KokkosConfigCommon.cmake.in
+++ b/cmake/KokkosConfigCommon.cmake.in
@@ -323,10 +323,13 @@ FUNCTION(kokkos_set_source_language LANG_SOURCE)
       IF(NOT DEFINED Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
         MESSAGE(FATAL_ERROR "Using language features requires Kokkos to be built with Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE=ON")
       ENDIF()
-      
-      IF(EXISTS ${LANG_SOURCE})
-        MESSAGE(DEBUG "KOKKOS: Setting language to ${Kokkos_COMPILE_LANGUAGE} for SOURCE ${LANG_SOURCE}")
-        SET_PROPERTY(SOURCE ${LANG_SOURCE} PROPERTY LANGUAGE ${Kokkos_COMPILE_LANGUAGE})
+
+      get_filename_component(LANG_SOURCE_FULL "${LANG_SOURCE}"
+                       REALPATH BASE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+
+      IF(EXISTS ${LANG_SOURCE_FULL})
+              MESSAGE(DEBUG "KOKKOS: Setting language to ${Kokkos_COMPILE_LANGUAGE} for SOURCE ${LANG_SOURCE_FULL}")
+        SET_PROPERTY(SOURCE ${LANG_SOURCE_FULL} PROPERTY LANGUAGE ${Kokkos_COMPILE_LANGUAGE})
       ELSE()
         MESSAGE(WARNING "KOKKOS: SOURCE `${LANG_SOURCE}` could not be found - Source language could not be set")
       ENDIF()
```
fixes that.